kalite/updates/download_track.py

```diff
@@ -0,0 +1,63 @@
+import json
+
+from django.conf import settings
+logging = settings.LOG
+
+from .settings import VIDEO_DOWNLOAD_QUEUE_FILE
+
+class VideoQueue:
+    """
+    This class handles queueing of videos for download.
+    It is written to by functions that manipulate the queue,
+    and read from by the videodownload management command that
+    ultimately downloads the videos.
+    """
+
+    def __init__(self):
+
+        self.load()
+
+    def add_files(self, files, language=None):
+        """
+        Add files to the queue - this should be a list of youtube_ids
+        and optionally, the language of the video.
+        """
+        files = [{"youtube_id": filename, "language": language} for filename in reversed(files)]
+        files.extend(self.queue)
+        self.queue = files
+        self.save()
+
+    def save(self):
+        """Save the current queue to disk"""
+        with open(VIDEO_DOWNLOAD_QUEUE_FILE, "w") as f:
```

Comment:
What happens if several processes or threads try to access this file simultaneously?